kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java

```diff
@@ -580,36 +580,47 @@ protected void handleTopicMetadataRequest(KafkaHeaderAndRequest metadataHar,
+            getAllTopicsAsync().thenAccept((allTopicMap) -> {
+                if (allTopicMap.isEmpty()) {
+                    pulsarTopicsFuture.complete(allTopicMap);
+                    return;
+                }
+                AtomicInteger allTopicCount = new AtomicInteger(allTopicMap.size());
+                final Runnable completeOne = () -> {
+                    if (allTopicCount.decrementAndGet() == 0) {
+                        pulsarTopicsFuture.complete(topicMap);
+                    }
+                };
+
                 allTopicMap.forEach((topic, list) -> {
-                   list.forEach((topicName ->
-                           authorize(AclOperation.DESCRIBE, Resource.of(ResourceType.TOPIC, topicName.toString()))
-                           .whenComplete((authorized, ex) -> {
-                               if (ex != null || !authorized) {
-                                   allTopicMetadata.add(new TopicMetadata(
-                                           Errors.TOPIC_AUTHORIZATION_FAILED,
-                                           topic,
-                                           KopTopic.isInternalTopic(topicName.toString(), metadataNamespace),
-                                           Collections.emptyList()));
-                                   return;
-                               }
-                               topicMap.computeIfAbsent(
-                                       topic,
-                                       ignored -> Collections.synchronizedList(new ArrayList<>())
-                               ).add(topicName);
-                           })));
+                    list.forEach((topicName ->
+                            authorize(AclOperation.DESCRIBE, Resource.of(ResourceType.TOPIC, topicName.toString()))
+                                    .whenComplete((authorized, ex) -> {
+                                        if (ex != null || !authorized) {
+                                            allTopicMetadata.add(new TopicMetadata(
+                                                    Errors.TOPIC_AUTHORIZATION_FAILED,
+                                                    topic,
+                                                    KopTopic.isInternalTopic(topicName.toString(), metadataNamespace),
+                                                    Collections.emptyList()));
+                                            completeOne.run();
+                                            return;
+                                        }
+                                        topicMap.computeIfAbsent(
+                                                topic,
+                                                ignored -> Collections.synchronizedList(new ArrayList<>())
+                                        ).add(topicName);
+                                        completeOne.run();
+                                    })));
```

Comment:
Could you avoid the indent changes? I've compared carefully for a while and only found two lines (612 and 619) are actually needed.

Currently the indent style is not restricted. We should avoid any change about the existing indent except nearly the whole code block is changed.

Reply:
Addressed.